javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +520,356 @@ object Character {
   @inline def compare(x: scala.Char, y: scala.Char): Int =
     x - y
 
+  // UnicodeBlock
+
+  class Subset protected (name: String) {
+    final override def equals(that: Any): scala.Boolean = super.equals(that)
+    final override def hashCode: scala.Int = super.hashCode
+    final override def toString(): String = name
+  }
+
+  final class UnicodeBlock private (name: String, private val start: Int, private val end: Int) extends Subset(name)
+
+  object UnicodeBlock {
+    private[this] val allBlocks: ArrayList[UnicodeBlock] = new ArrayList[UnicodeBlock](220)
+    private[this] val blocksByNormalizedName = new java.util.HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replaceAll("\\s", ""), block)
+      //   tokens as _
+      blocksByNormalizedName.put(lower.replaceAll("[\\s\\-_]", "_"), block)
+    }
+
+    private[this] def addUnicodeBlock(name: String, properName: String, start: Int, end: Int): UnicodeBlock =  {
```

Comment:
Is the `name` ever different than `properName.toUpperCase().replaceAll("[ \\-]", "_")`? If not, we should derive `name` from `properName` to reduce the amount of generated code.

If yes, perhaps it's only for a few exceptions? In that case we could have an overload of `addUnicodeBlock` that derives `name` from `properName`, and use that as often as possible.

Reply:
👍 verified this is true, update incoming shortly